code/xcvm/lib/proto/src/lib.rs

```diff
@@ -0,0 +1,452 @@
+#![no_std]
```

Comment:
Are we using this mapping instead of `proto::X` etc? It's better to avoid having a dep to protobuf in the contracts themselves, we need to map from the proto representation of a program to the base, Rust one

Reply:
We can possibly convert every proto type to our SDK types but I don't think we should pay for this. It will unnecessarily increase the gas price and we already convert to rust types if we want to use any sdk function.

Reply:
You cannot be sure that the cost will be very high after optimization and this will lead to the contract knowing implementation details of the serialization and also force you to unwrap or error almost everywhere because of how protobuf is  encoded. Let's move off using protobuf types directly